StardewArchipelago/ModEntry.cs

```diff
@@ -369,6 +369,14 @@ private void OnDayStarted(object sender, DayStartedEventArgs e)
 
         private void DoBugsCleanup()
         {
+            if (_archipelago.SlotData.Mods.HasMod(ModNames.LUCK) && Game1.player.LuckLevel >= 8 && !_locationChecker.IsLocationChecked("Lucky Lunch Recipe"))
+            {
+                _locationChecker.AddCheckedLocation("Lucky Lunch Recipe");
+                if (!_archipelago.HasReceivedItem("Lucky Lunch Recipe"))

```

Comment:
Check if the player has the lucky lunch recipe before removing it. You can't remove a recipe that's not there.

But also, check if they're on chefsanity, otherwise you'll take it away from people who absolutely should have it